distributed/executor.py

```diff
@@ -976,6 +987,66 @@ def cancel(self, futures):
         return sync(self.loop, self._cancel, futures)
 
     @gen.coroutine
+    def _publish_dataset(self, data, name):
+        keys = [tokey(f.key) for f in futures_of(data)]
+        yield self.scheduler.publish_dataset(keys=keys, name=tokey(name),
+                                             data=pickle.dumps(data),
```

Comment:
Should probably use the `dumps` function rather than `pickle.dumps`.  It handles a few special cases.